pkg/reconciler/v1alpha1/taskrun/resources/pod.go

```diff
@@ -248,20 +285,56 @@ func MakePod(taskRun *v1alpha1.TaskRun, taskSpec v1alpha1.TaskSpec, kubeclient k
 			Annotations: annotations,
 			Labels:      makeLabels(taskRun),
 		},
-		Spec: corev1.PodSpec{
-			// If the build fails, don't restart it.
-			RestartPolicy:      corev1.RestartPolicyNever,
-			InitContainers:     initContainers,
-			Containers:         podContainers,
-			ServiceAccountName: taskRun.Spec.ServiceAccount,
-			Volumes:            volumes,
-			NodeSelector:       taskRun.Spec.NodeSelector,
-			Tolerations:        taskRun.Spec.Tolerations,
-			Affinity:           taskRun.Spec.Affinity,
-		},
+		Spec: podSpec,
 	}, nil
 }
 
+
+func mergeStepWithBaseContainer(base, override *corev1.Container) (*corev1.Container, error) {
+	if base == nil {
+		return override, nil
+	}
+
+	baseData, err := json.Marshal(base)
+	if err != nil {
+		return nil, err
+	}
+
+	overrideData, err := json.Marshal(override)
+	if err != nil {
+		return nil, err
+	}
+
+	emptyData, err := json.Marshal(&corev1.Container{})
+	if err != nil {
+		return nil, err
+	}
+
+	schema, err := strategicpatch.NewPatchMetaFromStruct(base)
+
+	if err != nil {
+		return nil, err
+	}
+
+	patch, err := strategicpatch.CreateThreeWayMergePatch(emptyData, overrideData, baseData, schema, true)
+	if err != nil {
+		return nil, err
+	}
+
+	mergedData, err := strategicpatch.StrategicMergePatchUsingLookupPatchMeta(baseData, patch, schema)
```

Comment:
would you mind adding some comments before each of the `strategicpatch` calls elaborating on what's going on (from my ignorant-of-strategic-merge point of view it seems weird that there are 3 different calls required here and im not sure about details like "why do we need 'using lookup patch meta'"

Reply:
...yeah, I’ll be honest, it’s a bit black magic to me - I just kept hitting things until I got the result I needed. =) But I got there right before wrapping up for the day, so I’ll add explanatory comments tomorrow.